ludwig/features/bag_feature.py

```diff
@@ -94,16 +88,14 @@ def add_feature_data(
 
 @register_input_feature(BAG)
 class BagInputFeature(BagFeatureMixin, InputFeature):
-    encoder = "embed"
-    vocab = []
-
-    def __init__(self, feature, encoder_obj=None):
-        super().__init__(feature)
-        self.overwrite_defaults(feature)
+    def __init__(self, input_feature_config: Union[BagInputFeatureConfig, Dict], encoder_obj=None, **kwargs):
+        input_feature_config = self.load_config(input_feature_config)
+        super().__init__(input_feature_config, **kwargs)
+        self.encoder_config = input_feature_config.encoder
```

Comment:
just a note to remind of the inconsistency of the self.encoder_config assignment in case there is a tied encoder and so encoder_obj is not None. This applies to all input feature types

Reply:
Great catch, I'm actually working on changing this right now which will address this comment and your first comment!

Reply:
This should be resolved now that the encoder and decoder configs live on the `encoder_obj` / `decoder_obj` now.